### PR TITLE
Create user info summary (mobile)

### LINKED ISSUE #3476: Create mobile responsive Colony Home Menu
## Specification

Relies on issue: https://github.com/JoinColony/colonyDapp/issues/3475

### Story

When navigating from the Colony Home page on my mobile, I would like the main hamburger menu to be mobile-optimised. 

### Description 

This issue addresses screen three of the [Figma design](https://www.figma.com/file/DjVCPq7LpkjEMEfQZtKgBdC1/Style-Guide?node-id=6643%3A9926).

### Design

**Main menu should adapt to mobile screens**
image

### This issue is complete when...

- [ ] Clicking on the hamburger icon results in a menu as seen in the design.
- [ ] Menu list items correctly link  to their destinations.
- [ ] The menu properly fits screens with width less than or equal to `768px`.

## Subtasks

- [ ] Wire up the hamburger menu
- [ ] Add relevant sections per design

### LINKED ISSUE #3480: Create Account summary menu on mobile
## Specification

Relates to: #3474 

### Story

When using the dapp on mobile, I would like to see a summary of my account information when tapping on my avatar.

### Description

This issue address Screen 7 of the [Figma design](https://www.figma.com/file/DjVCPq7LpkjEMEfQZtKgBdC1/Style-Guide?node-id=6654%3A9751).

### Design 

**When tapping the avatar, an account summary appears**
image

### This issue is complete when...

- [ ] An account summary appears on screens at or less than `768px` in width when tapping the avatar icon.

### PR DESCRIPTION
**Pull request by willm30**
## Description

This PR is for the creation and wiring up of a User info summary menu (accesible from User avatar). [Was formerly a larger PR that I've now split up, hence the incongruous branch name]

This relates to screen 3 of the [design](https://www.figma.com/file/DjVCPq7LpkjEMEfQZtKgBdC1/Style-Guide?node-id=6643%3A9926).

**New stuff** ✨

- [x]  `AvatarDropdownPopoverMobile` component

## Screenshots

**_Avatar popover_**

user-info

**_Manage tokens_**

manage-tokens

## TODO

- [x] Build popover 
- [x] Wire up manage tokens
- [x] Implement requested changes

Resolves #3480

### CODE REVIEW

**Comment by rdig** [comment]
@willm30 this is a tad hard to review since it also includes the commits from #3485 

Change it's base branch to the above PR 
Screenshot from 2022-06-17 14-55-37

And then it will only show the commits and diff for the changes you did specifically in this branch.

After that you have two ways to go about it:
1. merge it directly into the base PR _(#3485 and then merge the whole think into the _mobile responsiveness_ feature branch)_
2. change the base branch back, merge #3485, then rebase this, and merge it into the feature branch.

Up to you the way to go about it

**Comment by rdig** [comment]
Also, as a side note

You don't need to ping Arren seprately as pinging the _Frontend_ group will ping him as well :)

Screenshot from 2022-06-17 14-53-23

**Comment by willm30** [comment]
> Also, as a side note
> 
> You don't need to ping Arren seprately as pinging the _Frontend_ group will ping him as well :)
> 
> Screenshot from 2022-06-17 14-53-23

So obvious now that I think about it haha 😅

**Comment by willm30** [comment]
@rdig Ok thanks, have changed base branch.

File: src/modules/core/components/DropdownMenu/DropdownMenu.tsx [hunk 1]
```diff
@@ -11,12 +11,16 @@ interface Appearance {
...
+  className?: string;
 }
 
 const displayName = 'DropdownMenu';
 
-const DropdownMenu = ({ appearance, children, ...props }: Props) => (
-  <div className={getMainClasses(appearance, styles)} {...props}>
+const DropdownMenu = ({ appearance, className, children, ...props }: Props) => (
+  <div
+    className={`${getMainClasses(appearance, styles)} ${className}`}
```
Comment (rdig) [inline src/modules/core/components/DropdownMenu/DropdownMenu.tsx:21]: This is not a valid change.

I know why you did it, to able to style the component directly, without having to deal with resetting the various styles applied by themes.

But the way this was set up is the standard way most core components styles are set up. So please revert this, and just deal with resetting the styles where you need them

File: src/modules/core/components/DropdownMenu/DropdownMenuItem/DropdownMenuItem.tsx [hunk 2]
```diff
@@ -5,10 +5,11 @@ import styles from './DropdownMenuItem.css';
 const displayName = 'DropdownMenuItem';
 
 const DropdownMenuItem = ({
+  className,
   children,
   ...props
 }: HTMLAttributes<HTMLLIElement>) => (
-  <li className={styles.main} {...props} role="menuitem">
+  <li className={`${styles.main} ${className}`} {...props} role="menuitem">
```
Comment (rdig) [inline src/modules/core/components/DropdownMenu/DropdownMenuItem/DropdownMenuItem.tsx:12]: There's not theme here, so here, the correct course of action is to overwrite the class. _(so revert the change again)_

If you need to inherit styles for your overwrite, you can use the `composes:` declaration available from CSS Modules

File: src/modules/core/components/HamburgerMenu/HamburgerMenu.tsx [hunk 3]
```diff
@@ -1,16 +1,31 @@
...
+  const MenuLine = () => (
+    <div
+      className={classnames(styles.menuLine, {
+        [styles.menuOpen]: isOpen,
+      })}
+    />
+  );
+
   return (
     <div className={styles.container}>
```
Comment (rdig) [inline src/modules/core/components/HamburgerMenu/HamburgerMenu.tsx:21]: As a rule of thumb, root classes for a component are called `main`.

```suggestion
    <div className={styles.main}>
```

File: src/modules/core/components/Popover/PopoverSection/ColonySection.tsx [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+import { CREATE_COLONY_ROUTE } from '~routes/routeConstants';
+
+const MSG = defineMessages({
+  createColony: {
+    id: 'users.AvatarDropdown.AvatarDropdownPopover.link.createColony',
+    defaultMessage: 'Create a Colony',
+  },
+});
+
+const displayName = 'users.AvatarDropdown.AvatarDropdownPopover.ColonySection';
```
Comment (rdig) [inline src/modules/core/components/Popover/PopoverSection/ColonySection.tsx:14]: Wrong name

File: src/modules/core/components/Popover/PopoverSection/ColonySection.tsx [hunk 5]
```diff
@@ -0,0 +1,26 @@
+import React from 'react';
+import { defineMessages } from 'react-intl';
+import { DropdownMenuItem, DropdownMenuSection } from '~core/DropdownMenu';
+import NavLink from '~core/NavLink';
+import { CREATE_COLONY_ROUTE } from '~routes/routeConstants';
+
+const MSG = defineMessages({
+  createColony: {
+    id: 'users.AvatarDropdown.AvatarDropdownPopover.link.createColony',
```
Comment (rdig) [inline src/modules/core/components/Popover/PopoverSection/ColonySection.tsx:9]: Wrong id

File: src/modules/core/components/Popover/PopoverSection/ColonySection.tsx [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+    defaultMessage: 'Create a Colony',
+  },
+});
+
+const displayName = 'users.AvatarDropdown.AvatarDropdownPopover.ColonySection';
+
+const ColonySection = ({ itemClassName }: { itemClassName?: string }) => (
```
Comment (rdig) [inline src/modules/core/components/Popover/PopoverSection/ColonySection.tsx:16]: Please set up a `interface` for props, even if it's just one prop.

File: src/modules/core/components/Popover/PopoverSection/HelperSection.tsx [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+    id: 'users.AvatarDropdown.AvatarDropdownPopover.link.reportBugs',
+    defaultMessage: 'Report Bugs',
+  },
+  helpCenter: {
+    id: 'users.AvatarDropdown.AvatarDropdownPopover.link.helpCenter',
+    defaultMessage: 'Help Center',
+  },
+});
+
+const displayName = 'users.AvatarDropdown.AvatarDropdownPopover.HelperSection';
```
Comment (rdig) [inline src/modules/core/components/Popover/PopoverSection/HelperSection.tsx:20]: Wrong id's and names again

File: src/modules/core/components/Popover/PopoverSection/HelperSection.tsx [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+
+const HelperSection = ({ itemClassName }: { itemClassName?: string }) => (
```
Comment (rdig) [inline src/modules/core/components/Popover/PopoverSection/HelperSection.tsx:22]: Please set up a `interface` for props, even if it's just one prop.

File: src/modules/core/components/Popover/PopoverSection/HelperSection.tsx [hunk 9]
```diff
@@ -0,0 +1,43 @@
+import React from 'react';
+import { defineMessages } from 'react-intl';
+import { DropdownMenuItem, DropdownMenuSection } from '~core/DropdownMenu';
+import ExternalLink from '~core/ExternalLink';
+import { FEEDBACK, HELP } from '~externalUrls';
+
+import styles from '../../../../users/components/AvatarDropdown/AvatarDropdownPopover.css';
```
Comment (rdig) [inline src/modules/core/components/Popover/PopoverSection/HelperSection.tsx:7]:
```suggestion
import styles from '~users/AvatarDropdown/AvatarDropdownPopover.css';
```

File: src/modules/core/components/Popover/PopoverSection/MetaSection.tsx [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+import { ActionTypes } from '~redux/actionTypes';
+
+const MSG = defineMessages({
+  signOut: {
+    id: 'users.AvatarDropdown.AvatarDropdownPopover.link.signOut',
+    defaultMessage: 'Sign Out',
+  },
+});
+
+const displayName = 'users.AvatarDropdown.AvatarDropdownPopover.MetaSection';
```
Comment (rdig) [inline src/modules/core/components/Popover/PopoverSection/MetaSection.tsx:14]: Wrong name and Id's again

File: src/modules/core/components/Popover/PopoverSection/ColonySection.tsx (hunk 6 above)
Comment (rdig) [inline src/modules/core/components/Popover/PopoverSection/ColonySection.tsx:16]: This is not a core component, it cannot be shared with others. It's a `user` level component

File: src/modules/core/components/Popover/PopoverSection/MetaSection.tsx [hunk 11]
```diff
@@ -0,0 +1,32 @@
...
+
+const MetaSection = ({ itemClassName }: { itemClassName?: string }) => (
```
Comment (rdig) [inline src/modules/core/components/Popover/PopoverSection/MetaSection.tsx:16]: This is not a core component, it cannot be shared with others. It's a `user` level component

File: src/modules/core/components/Popover/PopoverSection/UserSection.tsx [hunk 12]
```diff
@@ -0,0 +1,69 @@
...
+
+interface Props {
+  colony?: Colony;
+  username?: Maybe<string>;
+  itemClassName?: string;
+}
+
+const displayName = 'users.AvatarDropdown.AvatarDropdownPopover.UserSection';
+
+const UserSection = ({ colony, username, itemClassName }: Props) => (
```
Comment (rdig) [inline src/modules/core/components/Popover/PopoverSection/UserSection.tsx:31]: Wrong id's, wrong name, and it's not supposed to be in `core`

File: src/modules/dashboard/components/ColonyHome/ColonyTitle/ColonyTitleHeading.tsx [hunk 13]
```diff
@@ -0,0 +1,42 @@
...
+  },
+});
+
+type Props = {
+  colony: Colony;
+};
+
+const displayName = 'dashboard.ColonyHome.ColonyTitle';
+
+const ColonyTitleHeading = ({
```
Comment (rdig) [inline src/modules/dashboard/components/ColonyHome/ColonyTitle/ColonyTitleHeading.tsx:22]: Didn't you add this already in #3485 ?

File: src/modules/dashboard/components/ColonyHome/ColonyTitle/ColonyTitleMenu.tsx [hunk 14]
```diff
@@ -0,0 +1,44 @@
...
+});
+
+type Props = {
+  colony: Colony;
+  children?: React.ReactChild;
+};
+
+const displayName = 'dashboard.ColonyHome.ColonyTitle';
+
+const ColonyTitleMenu = ({ colony: { colonyAddress }, children }: Props) => {
```
Comment (rdig) [inline src/modules/dashboard/components/ColonyHome/ColonyTitle/ColonyTitleMenu.tsx:24]: Didn't you add this already in https://github.com/JoinColony/colonyDapp/pull/3485 ?

File: src/modules/dashboard/components/HamburgerDropdown/HamburgerDropdownPopover.tsx [hunk 15]
```diff
@@ -0,0 +1,97 @@
...
+interface Props {
+  onlyLogout?: boolean;
+  closePopover: () => void;
+  colony: Colony;
+  colonyName: string;
+  username?: Maybe<string>;
+  isWalletConnected: boolean;
+}
+
+const HamburgerDropdownPopover = ({
```
Comment (rdig) [inline src/modules/dashboard/components/HamburgerDropdown/HamburgerDropdownPopover.tsx:47]: This should not be in `dashboard`, but in `users`

File: src/modules/pages/components/RouteLayouts/UserNavigation.tsx [hunk 16]
```diff
@@ -107,9 +112,23 @@ const UserNavigation = () => {
...
+  const isPendingBalanceZero = bigNumberify(
+    userLock?.pendingBalance || 0,
+  ).isZero();
 
   const { formatMessage } = useIntl();
 
+  const onlyLogout = !isNetworkAllowed;
+  const colony = colonyData?.processedColony as Colony;
+  const { username } = useLoggedInUser();
+
```
Comment (rdig) [inline src/modules/pages/components/RouteLayouts/UserNavigation.tsx:131]: I'm not OK with these values being in here. It puts too much load on the initial render. 

They should be nested under their respective components

File: src/modules/pages/components/RouteLayouts/UserNavigation.tsx [hunk 17]
```diff
@@ -253,10 +275,37 @@ const UserNavigation = () => {
...
+        <HamburgerDropdown
+          {...{
+            username,
+            walletAddress,
+            ethereal,
+            onlyLogout,
+            colony,
+            colonyName,
+            isWalletConnected,
+          }}
```
Comment (rdig) [inline src/modules/pages/components/RouteLayouts/UserNavigation.tsx:307]: This is just ridiculous

**Review by rdig — CHANGES_REQUESTED** [review]
This is PR is very undercooked. I stopped reviewing about 2 thirds down.

I've left comments, but after a while I just gave up as there's too much to address. It feels to me like this is still in progress work.

Also note that the colors of the Hamburger menu changed in the specs as well as the spacing for the token activation menu _(pay attention to the rounded corners as well)

**Comment by rdig** [comment]
@willm30 per the above review.

I think you're doing to much work per one PR. Split up your issues more. Less work per PR, easier to review, quicker turnaround.

**Comment by willm30** [comment]
@rdig Thanks for the feedback and sorry this branch isn't as polished as it should be! Completely agree that I took on too much at once and therefore lost sight of the details. 

I have some questions re the best way to make my corrections / improvements. 

1. This branch can be conceptually divided into at least two parts, the hamburger menu and the user info avatar popover. Should I create a new branch off this branch for each and submit those as PRs for merging with this branch? Each PR would thus be roughly half the size of this one.  Or should I directly push my changes to this branch but ensure I break things up better going forward? 

2. Re sharing of data. I agree that there are too many props being passed into the AvatarDropdown from the UserNavigation component. My thinking was that some of this data is also needed by the UserTokenActivationButton, so I wanted to pull it up to their first common ancestor. But you recommend instead to push them back down to where they were so the app doesn't have to fetch the data prematurely and so the props don't become unwieldy, even if it means copying the code that fetches the data in each component? 

3. Another design question I had was on the Menu Sections (UserSection, HelperSection, MetaSection etc.). In this branch, I extracted the sections into their own components so they could be used in both the AvatarDropdownPopover and the HamburgerDropdownPopover. I guess I'm applying the same thinking as 2. in the sense that I'm trying to reuse shared "stuff" (components in this case).  Is that how you'd have done it? 

4. Also, I think I'm slightly confused by the directory structure (i.e. what goes where and why), but I think this will come with time.

Apologies again for the massive PR and now a massive comment.  I'll keep things smaller going forward.

File: src/modules/core/components/DropdownMenu/DropdownMenu.css [hunk 18]
```diff
@@ -6,7 +6,7 @@
   display: block;
 }
 
-.baseStyles button {
+.baseStyles > ul > li > button {
```
Comment (willm30) [inline src/modules/core/components/DropdownMenu/DropdownMenu.css:9]: So that Manage Tokens button doesn't have its styles reset

File: src/modules/users/components/AvatarDropdown/AvatarDropdown.tsx [hunk 19]
```diff
@@ -15,20 +15,40 @@ import styles, {
...
+  userData: Partial<LoggedInUser>;
+  spinnerMsg: SimpleMessageValues;
+  userTokenData: UserTokenData;
+  appState: AppState;
 }
 
 const displayName = 'users.AvatarDropdown';
 
-const AvatarDropdown = ({ onlyLogout = false, colony }: Props) => {
-  const { username, walletAddress, ethereal } = useLoggedInUser();
```
Comment (willm30) [inline src/modules/users/components/AvatarDropdown/AvatarDropdown.tsx:29]: Instead of calling useLoggedInUser again (already called in parent), I pass down relevant variables in an object.

File: src/modules/pages/components/RouteLayouts/UserNavigation.tsx [hunk 20]
```diff
@@ -180,17 +184,19 @@ const UserNavigation = () => {
...
+              titleTextValues={{ isMobile: false }}
+            />
           </div>
         ) : (
           <div className={`${styles.elementWrapper} ${styles.walletWrapper}`}>
             {userCanNavigate && nativeToken && userLock && (
               <UserTokenActivationButton
-                nativeToken={nativeToken}
-                userLock={userLock}
+                userTokenData={getUserTokenData(userLock)}
```
Comment (willm30) [inline src/modules/pages/components/RouteLayouts/UserNavigation.tsx:196]: I've extracted the logic for getting user token data into a separate function which is being shared between this component and also `AvatarDropdownPopoverMobile`

File: src/modules/users/components/AvatarDropdown/AvatarDropdownPopoverMobile.tsx [hunk 21]
```diff
@@ -0,0 +1,164 @@
...
+              <TokenActivationPopover
+                activeTokens={activeBalance}
+                inactiveTokens={inactiveBalance}
+                totalTokens={totalBalance}
+                lockedTokens={lockedBalance}
+                token={nativeToken}
+                {...{ colony, walletAddress, isPendingBalanceZero }}
+              >
+                {({ toggle, ref }) => (
+                  <>
```
Comment (ArmandoGraterol) [inline src/modules/users/components/AvatarDropdown/AvatarDropdownPopoverMobile.tsx:142]: This empty bracket is not needed

File: src/modules/users/components/UserTokenActivationButton/UserTokenActivationButton.tsx [hunk 22]
```diff
@@ -1,15 +1,15 @@
...
-import { getFormattedTokenValue } from '~utils/tokens';
-import Numeral from '~core/Numeral';
-import { FullColonyFragment, UserLock, UserToken } from '~data/index';
+
+import { FullColonyFragment } from '~data/index';
 import { Address } from '~types/index';
 
 import styles from './UserTokenActivationButton.css';
+import UserTokenActivationDisplay from './UserTokenActivationDisplay';
+import { UserTokenBalanceData } from '~types/tokens';
```
Comment (ArmandoGraterol) [inline src/modules/users/components/UserTokenActivationButton/UserTokenActivationButton.tsx:12]: Same with these imports, they should be higher grouped with similar imports as them, and the CSS import last

File: src/modules/users/components/UserTokenActivationButton/UserTokenActivationDisplay.tsx [hunk 23]
```diff
@@ -0,0 +1,42 @@
+import React from 'react';
+import { BigNumber } from 'ethers/utils';
+
+import { getFormattedTokenValue } from '~utils/tokens';
+import Numeral from '~core/Numeral';
+
+import styles from './UserTokenActivationButton.css';
+import { UserToken } from '~data/generated';
```
Comment (ArmandoGraterol) [inline src/modules/users/components/UserTokenActivationButton/UserTokenActivationDisplay.tsx:8]: Same with this import

File: src/modules/users/components/UserTokenActivationButton/UserTokenActivationDisplay.tsx [hunk 24]
```diff
@@ -0,0 +1,42 @@
...
+  const formattedTotalBalance = getFormattedTokenValue(
+    totalBalance,
+    nativeToken?.decimals,
+  );
+
+  return (
+    <div>
+      <span
+        className={`${styles.dot} ${
+          (inactiveBalance.gt(0) || totalBalance.isZero()) && styles.dotInactive
```
Comment (ArmandoGraterol) [inline src/modules/users/components/UserTokenActivationButton/UserTokenActivationDisplay.tsx:33]: We have a library to conditionally add styles like you are doing here but in a more readable way, it's called `classnames`. You can search for examples in the dapp by looking for `import classnames` :)

**Review by ArmandoGraterol — COMMENTED** [review]
Nice work Will! I just noticed some small things that could need a change but after that, this should be good to go from my side

File: src/modules/users/components/AvatarDropdown/AvatarDropdownPopoverMobile.css [hunk 25]
```diff
@@ -0,0 +1,31 @@
...
+  align-items: center;
+  padding: 20px 10px;
+  color: var(--temp-grey-blue-7);
+}
+
+.itemChild {
+  padding: 0px 6px;
+  height: 100%;
+  border-radius: 4px;
+  background-color: rgba(118, 116, 139, 0.1);
```
[the remark this answers is not in the document]
Comment (willm30) [reply src/modules/users/components/AvatarDropdown/AvatarDropdownPopoverMobile.css:19]: I'd not seen `color-mod` before -- great find!

File: src/modules/users/components/AvatarDropdown/AvatarDropdown.tsx [hunk 26]
```diff
@@ -15,20 +15,38 @@ import styles, {
   horizontalOffset,
   verticalOffset,
 } from './AvatarDropdown.css';
+import AvatarDropdownPopoverMobile from './AvatarDropdownPopoverMobile';
+import { SimpleMessageValues } from '~types/index';
+import { UserTokenBalanceData } from '~types/tokens';
```
[the remark this answers is not in the document]
Comment (willm30) [reply src/modules/users/components/AvatarDropdown/AvatarDropdown.tsx]: ☹️

**Review by ArmandoGraterol — APPROVED** [review]
Looking good! I just noticed some small things and made some suggestions but other than that this is good to go :)

File: src/modules/users/components/AvatarDropdown/AvatarDropdownPopoverMobile.tsx [hunk 27]
```diff
@@ -0,0 +1,158 @@
...
+
+import Button from '~core/Button';
+import { Colony } from '~data/index';
+import DropdownMenu, {
+  DropdownMenuSection,
+  DropdownMenuItem,
+} from '~core/DropdownMenu';
+import MaskedAddress from '~core/MaskedAddress';
+import MemberReputation from '~core/MemberReputation';
+import UserTokenActivationDisplay from '../UserTokenActivationButton/UserTokenActivationDisplay';
```
[the remark this answers is not in the document]
Comment (willm30) [reply src/modules/users/components/AvatarDropdown/AvatarDropdownPopoverMobile.tsx:12]: Aren't the `~core/abc` syntax and `../xyz` syntax both examples of relative imports?

[the remark this answers is not in the document]
Comment (willm30) [reply src/modules/users/components/AvatarDropdown/AvatarDropdownPopoverMobile.tsx:12]: Or `~core` is absolute? And therefore should be below the `../`?

[the remark this answers is not in the document]
Comment (rdig) [reply src/modules/users/components/AvatarDropdown/AvatarDropdownPopoverMobile.tsx:12]: Core should always be first when dealing with component imports.

Here's a rule of thumb for import priorities:
1. core
2. user
3. dashboard
4 (last). local imports (same folder as current component)

_(page should be before local, but you'll rarely have to deal with that)_

**Review by chinins — APPROVED** [review]
Very nice work. Other than Armando's comments, all good.

File: src/modules/users/components/AvatarDropdown/AvatarDropdownPopoverMobile.tsx (hunk 27 above)
[the remark this answers is not in the document]
Comment (ArmandoGraterol) [reply src/modules/users/components/AvatarDropdown/AvatarDropdownPopoverMobile.tsx:12]: Yeah, my bad for the misunderstanding but it's as Raul said.  We normally order these imports based on location and if they are relative or absolute, so in this case the import of `UserTokenActivationDisplay` is quite a local import compared to the imports of `~core` and `~types` that you have here, and the more local an import is, the more it should stay below

For this case, I would put this import between the import of `TokenActivationPopover` (or just above,  they have the same level of "localness" after all) and `AppState` because it's "as local as" the import `TokenActivationPopover` but not too direct like `AppState`, which is imported from the same folder.

File: src/modules/users/components/HamburgerDropdown/HamburgerDropdownPopover.tsx [hunk 28]
```diff
@@ -0,0 +1,98 @@
...
+                </DropdownMenuItem>
+              </DropdownMenuSection>
+            )}
+            <UserSection colony={colony} username={username} />
+            <ColonySection />
+            <HelperSection />
+            {isWalletConnected && <MetaSection />}
+          </>
+        ) : (
+          isWalletConnected && <MetaSection />
```
[the remark this answers is not in the document]
Comment (willm30) [reply src/modules/users/components/HamburgerDropdown/HamburgerDropdownPopover.tsx]: Yes good idea, thanks.

File: src/modules/users/components/PopoverSection/UserSection.tsx [hunk 29]
```diff
@@ -0,0 +1,70 @@
...
+    )}
+    {username && (
+      <DropdownMenuItem>
+        <NavLink
+          to={USER_EDIT_ROUTE}
+          text={MSG.settings}
+          data-test="userProfileSettings"
+        />
+      </DropdownMenuItem>
+    )}
```
[the remark this answers is not in the document]
Comment (willm30) [reply src/modules/users/components/PopoverSection/UserSection.tsx]: Also agree! Thanks.

File: src/modules/users/components/AvatarDropdown/AvatarDropdownPopoverMobile.tsx (hunk 27 above)
[the remark this answers is not in the document]
Comment (willm30) [reply src/modules/users/components/AvatarDropdown/AvatarDropdownPopoverMobile.tsx]: I've grouped the `../` and put them below the `~xx` as I have seen this elsewhere in the codebase.

[the remark this answers is not in the document]
Comment (willm30) [reply src/modules/users/components/AvatarDropdown/AvatarDropdownPopoverMobile.tsx]: Ok thanks both for clarifying. Wrote the above before I saw your comments but I got the jist of it anyhow
